utils/yoke/src/yokeable.rs

```diff
@@ -259,18 +275,22 @@ where
         debug_assert!(mem::size_of::<Cow<'a, T>>() == mem::size_of::<Self>());
         let ptr: *const Self = (&from as *const Self::Output).cast();
         mem::forget(from);
-        core::ptr::read(ptr)
+        // Safety: `ptr` is certainly valid, aligned and points to a properly initialized value, as
+        // it comes from a mem::forget-ed
+        unsafe { core::ptr::read(ptr) }
     }
     #[inline]
     fn transform_mut<F>(&'a mut self, f: F)
     where
         F: 'static + for<'b> FnOnce(&'b mut Self::Output),
     {
         // Cast away the lifetime of Self
+        // Safety: see the documentation of the trait method.
```

Comment:
I would explicitly document on the trait method that the implementation is expected to be equivalent to `f(transmute(self))`, and then here say that this matches the requirement of the trait which is to implement this this way